swagger_templates/python/api_client.mustache

```diff
@@ -95,55 +108,46 @@ class ApiClient(object):
     def __call_api(self, resource_path, method,
                    path_params=None, query_params=None, header_params=None,
                    body=None, post_params=None, files=None,
-                   response_type=None, auth_settings=None, callback=None,
-                   _return_http_data_only=None, collection_formats=None):
+                   response_type=None, auth_settings=None, callback=None):
 
-        # header parameters
+        # headers parameters
         header_params = header_params or {}
         header_params.update(self.default_headers)
         if self.cookie:
             header_params['Cookie'] = self.cookie
         if header_params:
             header_params = self.sanitize_for_serialization(header_params)
-            header_params = dict(self.parameters_to_tuples(header_params,
-                                                           collection_formats))
 
         # path parameters
         if path_params:
             path_params = self.sanitize_for_serialization(path_params)
-            path_params = self.parameters_to_tuples(path_params,
-                                                    collection_formats)
-            for k, v in path_params:
-                v_str = str(v)
+            for k, v in iteritems(path_params):
+                v_str = str(self.to_path_value(v))
                 # Check for the special case of the
                 # x-isi-url-encode-path-param, which indicates that the
                 # parameter should be encoded with quote_plus in order
                 # to encode the '/' character.
-                quote_plus_tag = "__x-isi-url-encode-path-param__"
-                tag_len = len(quote_plus_tag)
                 # check if the first part of v_str matches the tag
-                if v_str[:tag_len] == quote_plus_tag:
+                if v_str[:self.quote_plus_tag_len] == self.quote_plus_tag:
                     # remove "__x-isi-url-encode-path-param__"
-                    v_str = v_str[tag_len:]
+                    v_str = v_str[self.quote_plus_tag_len:]
                     # then url-encode with quote_plus
-                    resource_path = resource_path.replace(
-                        '{%s}' % k, quote_plus(v_str))
+                    replacement = quote_plus(v_str)
                 else:
-                    resource_path = resource_path.replace(
-                        '{%s}' % k, quote(v_str))
+                    replacement = quote(v_str)
+                resource_path = resource_path.\
+                    replace('{' + k + '}', replacement)
 
         # query parameters
         if query_params:
             query_params = self.sanitize_for_serialization(query_params)
-            query_params = self.parameters_to_tuples(query_params,
-                                                     collection_formats)
+            query_params = {k: self.to_path_value(v)
```

Comment:
These other changes are just due to the fact that I used the wrong template in my last commit.